sources/packages/backend/libs/integrations/src/services/disbursement-schedule/e-cert-processing-steps/persist-calculations-step.ts

```diff
@@ -33,14 +34,42 @@ export class PersistCalculationsStep implements ECertProcessStep {
     log.info("Saving all e-Cert calculations.");
     const now = new Date();
     const disbursement = eCertDisbursement.disbursement;
-    disbursement.disbursementScheduleStatus =
-      DisbursementScheduleStatus.ReadToSend;
-    disbursement.readyToSendDate = now;
-    disbursement.modifier = this.systemUsersService.systemUser;
-    disbursement.updatedAt = now;
-    await entityManager.getRepository(DisbursementSchedule).save(disbursement);
+    // Persists the changes to the disbursement.
+    // Using update instead save for better performance.
+    const disbursementScheduleRepo =
+      entityManager.getRepository(DisbursementSchedule);
+    disbursementScheduleRepo.update(
+      { id: disbursement.id },
+      {
+        disbursementScheduleStatus: DisbursementScheduleStatus.ReadyToSend,
+        readyToSendDate: now,
+        tuitionRemittanceEffectiveAmount:
+          disbursement.tuitionRemittanceEffectiveAmount,
+        modifier: this.systemUsersService.systemUser,
+        updatedAt: now,
+      },
+    );
+    // Persist the changes to the disbursement values.
+    // Using update instead save for better performance.
+    const disbursementValueRepo =
+      entityManager.getRepository(DisbursementValue);
+    const disbursementValuesUpdatesPromises =
+      disbursement.disbursementValues.map((disbursementValue) => {
+        return disbursementValueRepo.update(
+          { id: disbursementValue.id },
+          {
+            overawardAmountSubtracted:
+              disbursementValue.overawardAmountSubtracted,
+            restrictionAmountSubtracted:
+              disbursementValue.restrictionAmountSubtracted,
+            restrictionSubtracted: disbursementValue.restrictionSubtracted,
+            effectiveAmount: disbursementValue.effectiveAmount,
```

Comment:
Thanks for making the change. I wanted to mention about the typeorm interceptors/listeners which acts on insert update operations. Should we consider having the interceptors for workers and queue consumers now  because we had a struggle earlier in typeorm that it was not able to read the http context and get the user object. But when it comes to workers and Queue consumers it is going to be the system user service object.